puzzlegenerator/puzzlegenerator_ui/src/main/java/com/flurnamenpuzzle/generator/ui/view/FieldNameMapSelectionCard.java

```diff
@@ -123,16 +144,49 @@ public void actionPerformed(ActionEvent e) {
 		chooseFieldNameButton.addActionListener(new ActionListener() {
 			@Override
 			public void actionPerformed(ActionEvent e) {
-				shapeFileChooser.showOpenDialog(null);
-				fieldNamePath.setText(shapeFileChooser.getSelectedFile().getPath());
+				int returnVal = shapeFileChooser.showOpenDialog(null);
+				// check if user has selected a file
+				if (returnVal == JFileChooser.APPROVE_OPTION) {
+					File selectedFile = shapeFileChooser.getSelectedFile();
+					String pathOfSelectedFile = selectedFile.getPath();
+					fieldNamePath.setText(pathOfSelectedFile);
+					String extension = controller.getFileExtension(selectedFile);
+					if(extension.equals("shp")){
+						notificationLabel.setForeground(PuzzleGeneratorConfig.SUCCESS_COLOR);
+						notificationLabel.setText("Shape Datei ausgewählt.");
+					}else{
+						notificationLabel.setForeground(PuzzleGeneratorConfig.FAIL_COLOR);
+						notificationLabel.setText("Es wurde keine Shape Datei ausgewählt.");
+					}
+				}
 			}
 		});
 
 		chooseMapButton.addActionListener(new ActionListener() {
 			@Override
 			public void actionPerformed(ActionEvent e) {
-				tifFileChooser.showOpenDialog(null);
-				mapPath.setText(tifFileChooser.getSelectedFile().getPath());
+				int returnVal = tifFileChooser.showOpenDialog(null);
+				// check if user has selected a file
+				if (returnVal == JFileChooser.APPROVE_OPTION) {

```

Comment:
Duplizierter Code: Die Zeilen 170-181 und 149-160 tun inhaltlich dasselbe. Das kann in eine separate Methode ausgelagert werden.